### ISSUE TITLE
ofxDLib, ofxLearn, ofxFaceTracker

### ISSUE THREAD

**Issue by genekogan** [roymacdonald/ofxDLib#3]
this looks awesome roy :+1:  :)

i've been meaning to update ofxLearn to the newest dlib. you and i might be duplicating work -- is there any reason you didn't build this on top of ofxLearn (since it already has a bunch of other dlib wrappers).

also to my eyes the facetracker is not as robust as ofxFaceTracker. see 
<img width="1260" alt="screen shot 2016-03-27 at 10 05 18 pm" src="https://cloud.githubusercontent.com/assets/1335251/14069495/a3d4ea80-f468-11e5-992f-1337f26841f4.png">

also have you seen [this](http://blog.dlib.net/2014/08/real-time-face-pose-estimation.html).
does real-time facetracking with dlib which appears super robust, looks like there is an example included, have you seen that?

**Comment by roymacdonald** [comment]
Hey @genekogan Thanks!
Actually there's no reason for not having using ofxLearn as a starting point. All this begun when andy dayton was asking in the OF forum how to correctly add dlib in an of project in order to use the face tracker. I can't recall why I didn't use ofxLearn at that point, maybe because you were not using the latest version. I posted that project in the of forum and several people got interested. This addon is just porting that project into an addon and making it work correctly with PG.

Sure, we must unite forces. It would be a good thing to get ofxLearn and this to work together.
I haven't tried anything else . I've been quite busy lately thanks to Paul, here in Shanghai. ;)
I'll let you know if there's any progress.
best!

**Comment by antimodular** [comment]
@genekogan about the link you posted to dlib face tracker.
isn't that what roy did with the ofxDLib face tracker example:
https://github.com/roymacdonald/ofxDLib/blob/master/src/FaceTracker.cpp#L15

**Comment by ofZach** [comment]
@genekogan I found this https://github.com/TadasBaltrusaitis/CLM-framework which sits on top of dlib to be significantly better then ofxFaceTracker....  haven't tested dlib directly but CLM is great.    (also in my experience CLAHE helps alot with face trackers http://imagej.net/Enhance_Local_Contrast_(CLAHE))

**Comment by genekogan** [comment]
@ofZach i think @kylemcdonald has been playing with dlib for facetracking recently (https://twitter.com/kcimc/status/719563059673579520), not sure if he has tried CLM. it does look quite promising

**Comment by antimodular** [comment]
something to consider is that from my experience dlib gets pretty slow when processing images/video at higher resolutions 960x720 is almost too big already.
while i seem to remember ofxFaceTracker working well at HD res.

**Comment by ofZach** [comment]
yes I had to scale down the image I was passing into to CLM, I think it fine at 640x480 so it was something in that size range.   Faces couldn't be smaller than about 70 pixels but that worked for my purposes.   I think that might have to do with HoG vs haar for locating the face.

**Comment by stephanschulz** [comment]
dlib version 19.2 is out with neural net face detection + etc. 
here is my very first attempt at making it run. it runs very slow and not very good.
https://github.com/antimodular/ofxDLib/tree/development_dlib19.2/example-FaceTracker_DNN

**Comment by kylemcdonald** [comment]
i was testing dlib on my mbp and on an nvidia gtx780 with gpu acceleration and i was getting 3-5fps for images in the 800x800 to 1600x1600 range. and the results are perfect in terms of accuracy. i'm testing with modified example script that just outputs json for a list of images fed in via the command line: https://gist.github.com/kylemcdonald/b88f99d1d97f016a686c11bd378f3b5c
